app/Group.php

```diff
@@ -251,28 +251,46 @@ public function getGroupStats($emissionRatio)
         $participants = 0;
         $hours_volunteered = 0;
         $co2 = 0;
-        $waste = 0;
-
-        foreach ($allPastParties as $party) {
-            $partyco2 = 0;
-            $participants += $party->pax;
-            $hours_volunteered += $party->hoursVolunteered();
-
-            foreach ($party->devices as $device) {
-                if ($device->isFixed()) {
-                    $partyco2 += $device->co2Diverted($emissionRatio, $Device->displacement);
-                    $waste += $device->ewasteDiverted();
-                }
-            }
-            $co2 += $partyco2;
+        $ewaste = 0;
+        $unpowered_waste = 0;
+        $fixed_devices = 0;
+        $fixed_powered = 0;
+        $fixed_unpowered = 0;
+        $repairable_devices = 0;
+        $dead_devices = 0;
+        $no_weight = 0;
+        $devices_powered = 0;
+        $devices_unpowered = 0;
+
+        foreach ($allPastParties as $event) {
+            $eventStats = $event->getEventStats($emissionRatio);
+
+            $co2 += $eventStats['co2'];
+            $participants += $eventStats['participants'];
+            $hours_volunteered += $event->hoursVolunteered();
+            $ewaste += $eventStats['ewaste'];
+            $unpowered_waste += $eventStats['unpowered_waste'];
+
+            $fixed_devices += $eventStats['fixed_devices'];
+            $fixed_powered += $eventStats['fixed_powered'];
+            $fixed_unpowered += $eventStats['fixed_unpowered'];
+            $repairable_devices += $eventStats['repairable_devices'];
+            $dead_devices += $eventStats['dead_devices'];
+            $no_weight += $eventStats['no_weight'];
+            $devices_powered += $eventStats['devices_powered'];
+            $devices_unpowered += $eventStats['devices_unpowered'];
         }
```

Comment:
Are there any stats on the event which we do _not_ want to sum for the group?  We could just iterate through $eventStats rather than list them individually.

Reply:
I'll double check them - I don't think there's any event stats we need to exclude from the group stats.  

There's one currently not being included in the event stats, hours volunteered - but would make sense to add this to event stats, and then yes we could iterate through $eventStats.

Reply:
I've changed this now to iterate through the keys from the event stats.